documentation/puppet_server_metrics_performance.markdown

```diff
@@ -50,9 +52,11 @@ If you are setting up Puppet Server for the first time, start by increasing your
 
 If you must add JRubies, remember that Puppet Server is tuned by default to use one fewer than your total number of CPUs, with a maximum of 4 CPUs, for the number of available JRubies. You can change this by setting `max-active-instances` in [`puppetserver.conf`][puppetserver.conf], under the `jruby-puppet` section.
 
+> **Note:** The following applies only to the normal mode where increasing `max-active-instances` creates whole independent JRuby instances. In multithreaded mode, this setting instead controls the number of threads that the single JRuby instance will process concurrently, and therefore has different scaling characteristics. Tuning recommendations for this mode are under development, see [SERVER-2823](https://tickets.puppetlabs.com/browse/SERVER-2823).
```

Comment:
Why not put this note above the first paragraph of this section?

Reply:
Yes I agree. And also starting with "Adding more JRubies applies only to...etc". Instead of "The following...".

Reply:
If you're considering "a JRuby" to just as easily mean "a JRuby thread", the first paragraph applies to both modes equally. We use the same defaults and control the number with the same setting. But I'll rework this a bit, into that first paragraph. I still want to indicate that all the stuff in the paragraphs below about resourcing doesn't apply to threaded mode, but we don't know what to recommend instead yet.

Reply:
I don't love considering "a JRuby" to be an instance or a thread. I think, since we've already been referring to instances as "JRubies", we should be explicit when referring to threads.

I think it's much clearer now. 👍